ldk/javascript/src/network/index.ts

```diff
@@ -29,6 +28,57 @@ export interface HTTPResponse {
   headers: Record<string, string[]>;
 }
 
+/**
+ * a simplified representation of a callback which take error
+ */
+export type CallbackError = (error: Error | undefined) => void;
+
+/**
+ * configuration object to configure a websocket handshake
+ */
+export interface SocketConfiguration {
+  /**
+   * websocket server endpoint url: '{schema}://{host}:{port}' - schema could be ws or wss
+   */
+  url: string;
+  /**
+   * collection of the handshake headers
+   */
+  headers?: Record<string, string[]>;
+  /**
+   * specifies if compression is used
+   */
+  useCompression?: boolean;
+  /**
+   * specifies the client's requested subprotocols.
+   */
+  subprotocols?: Array<string>;
+}
+
+/**
+ * Object to communicate with the websocket
+ */
+export interface Socket {
+  /**
+   * writes message to a websocket
+   * @param message text or data message
+   * @param callback function to call if error occured
+   */
+  writeMessage(message: string | Uint8Array, callback: CallbackError): Promise<void>;
+  /**
+   * closes websocket
+   * @param callback function to call if error occured
+   */
+  close(callback: CallbackError): Promise<void>;
+  /**
+   * allows to listen for a websocket message (there should be only 1 listen message callback per socket to allow messages being fully received)
```

Comment:
I would update

"there should be only 1 listen message callback per socket to allow messages being fully received"

to

"there **must** be only one listener registered per socket for messages to be fully received"

Reply:
"In the end, there can be only one." - Accepted!

Reply:
done